### PR TITLE
Config files restored + BIN enhancements + BIN/README sync

### PR DESCRIPTION
**Pull request by SkyCrawl**
**EDIT:**

What I've changed:

## ./config.rb, ./config.ru

These were added, with pointers to the wiki.

## ./bin/gollum

1. Added `opts.program_name = "gollum"`, just in case.
2. Options were divided into "categories", according to significance/purpose.
3. Deleted the `--gollum-path` option since it was completely redundant (in favour of ARGV).
4. Added short style options to several most important/used options.
4. Fixed several potential bugs and confusions:
    * `--port` now raises an error if the arg is not an integer number.
    * `--base-path` is now escaped before being used. While setting base path to a URL that needs not escaping STILL WORKS, the contrary DOESN'T WORK because there is an infinite redirect (**very probably another bug somewhere down the road**). To reproduce:
        * `gollum -b "//wiki"`
        * `http://localhost:4567`
    * `--user-icons` now only accepts the defined values.
5. `gollum --help`
    * Fixed alignment and indentation.
        * All lines are now indented with 2 spaces so that the user can easily tell (visually) where the printed message begins.
        * Everything is now aligned with everything so the message should now look very nice (see the image below).
        * The message's requirement to terminal width has been reduced quite a bit, and height requirement has consequently been increased a bit (see the images below).
    * Added a backlink to https://github.com/gollum/gollum.
6. Added a Rack redirect case to get rid of a "not found" error when testing an escaped base path (see above).
7. Tweaked the IRB welcome message a little and added a backlink to https://github.com/gollum/gollum-lib.

## ./bin/gollum, README

I forgot to update `./bin/gollum` accordingly when I rewrote the README so the new commits address that as well. Executable and README are now synced together, with the exception of a handful of phrases and the lack of said option categories in README (where they would only make things worse).

## Comparison of `gollum --help` [IMAGES]

[Original](https://dl.dropboxusercontent.com/u/100302745/gollum_pics/bin-help_original.png).

[New](https://dl.dropboxusercontent.com/u/100302745/gollum_pics/bin-help_new.png).

----

P.S.: oh, and a small question out of curiosity... is there a reason the `History.md` file is kept? Personally, I think such information belongs into "additional information" fields of respective tags :).

### CODE REVIEW

**Comment by dometto** [comment]
Looks good to me! @bartkamphorst, any suggestions?

**Comment by SkyCrawl** [comment]
More stuff added/changed, I've updated my original post. Sorry for confusion and please pardon me for not waiting until this PR is merged :).

**Comment by dometto** [comment]
Like the examples in the bin :+1: One issue, `bundle` is giving me this warning:
```
gollum at /Users/dawa/Code/gollum_project/gollum did not have a valid gemspec.
This prevents bundler from installing bins or native extensions, but that may not affect its functionality.
The validation message from Rubygems was:
  ["docs/sanitization.md"] are not files
```
Maybe remove `docs/sanitization.md` from the gemspec manifest?

File: README.md [hunk 1]
```diff
@@ -125,37 +126,36 @@ Gollum comes with the following command line options:
 | ----------------- | --------- | ----------- |
 | --host            | [HOST]    | Specify the hostname or IP address to listen on. Default: `0.0.0.0`.<sup>1</sup> |
 | --port            | [PORT]    | Specify the port to bind Gollum with. Default: `4567`. |
-| --gollum-path     | [PATH]    | Specify path to the Gollum git repository. Default: current working directory. |
-| --config          | [CONFIG]  | Path to the configuration file. |
+| --config          | [FILE]  | Specify path to the Gollum's configuration file. |
```
Comment (dometto) [inline README.md]: Remove 'the'

**Comment by dometto** [comment]
Stylewise, I'm also for writing 'gollum' in lowercase instead of 'Gollum', although I don't feel very strongly.

File: README.md (hunk 1 above)
Comment (SkyCrawl) [inline README.md]: Done :+1:

**Comment by SkyCrawl** [comment]
> Maybe remove docs/sanitization.md from the gemspec manifest?

Done. Sorry about that :).

> Stylewise, I'm also for writing 'gollum' in lowercase instead of 'Gollum', although I don't feel very strongly.

Me neither. "Gollum" is often used to denote a "term" like "Gollum page" though, so maybe the uppercase is a little better (in my eyes anyway)? But I don't really care either :).

**Comment by dometto** [comment]
Sorry for the long wait @SkyCrawl!

**Pull request merged**